src/modules/jcms_rest/src/Plugin/rest/resource/JobAdvertListRestResource.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Drupal\jcms_rest\Plugin\rest\resource;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\jcms_rest\Response\JCMSRestResponse;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Provides a resource to get view modes by entity and bundle.
+ *
+ * @RestResource(
+ *   id = "job_advert_list_rest_resource",
+ *   label = @Translation("Job advert list rest resource"),
+ *   uri_paths = {
+ *     "canonical" = "/job-adverts"
+ *   }
+ * )
+ */
+class JobAdvertListRestResource extends AbstractRestResourceBase {
+
+  /**
+   * Responds to GET requests.
+   *
+   * Returns a list of bundles for specified entity.
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+   *   Throws exception expected.
+   *
+   * @todo - elife - nlisgo - Handle version specific requests
+   */
+  public function get() {
+    $base_query = \Drupal::entityQuery('node')
+      ->condition('status', NodeInterface::PUBLISHED)
+      ->condition('changed', \Drupal::time()->getRequestTime(), '<')
+      ->condition('type', 'job_advert');
+
+    $this->filterShow($base_query);
```

Comment:
The `filterShow` method will need some work so it can work for `events` and `job_adverts`. Currently, it is only working for `events`.